src/strategies/predefined.zsh

```diff
@@ -0,0 +1,99 @@
+# zsh-augosuggestion: predefined strategy
+# 1. search in history, returns result if it exists
+# 2. search in predefined files
+#
+# configuration:
+# 
+# ZSH_AUTOSUGGEST_PREDEFINE_NAME  - auto generated predefine file name
+# ZSH_AUTOSUGGEST_PREDEFINE_PATH  - user defined files separated by semicolon
+#
+# "$HOME/.zsh_autosuggest" will be generated at the first time
```

Comment:
Please don't default to dotfile in top level of home directory. [XDG Base Dir spec](https://specifications.freedesktop.org/basedir-spec/basedir-spec-latest.html) is more than a decade old and equally applies to GUI applications and CLI utilities.

`$XDG_CACHE_HOME` is a perfect location for this kind of file.

Reply:
`$HOME/.zsh_autosuggest` has been changed to `$XDG_DATA_HOME/zsh_autosuggest/predefined`, or `~/.local/share/zsh_autosuggest/predefined` if `$XDG_DATA_HOME` is empty.